specs/core/0_beacon-chain.md

```diff
@@ -2310,6 +2327,44 @@ def process_attester_slashing(state: BeaconState,
         slash_validator(state, index)
 ```
 
+##### Randao reveal slashings
+
+Verify that `len(block.body.randao_reveal_slashings) <= MAX_RANDAO_REVEAL_SLASHINGS`.
+
+For each `randao_reveal_slashing` in `block.body.randao_reveal_slashings`, run the following function:
+
+```python
+def process_randao_reveal_slashing(state: BeaconState,
+                                   randao_reveal_slashing: RandaoRevealSlashing) -> None:
+    """
+    Process ``RandaoRevealSlashing`` operation.
+    Note that this function mutates ``state``.
+    """
+    revealer = state.validator_registry[randao_reveal_slashing.revealer_index]
+    masker = state.validator_registry[randao_reveal_slashing.masker_index]
+    pubkeys = [revealer.pubkey, masker.pubkey]
+    message_hashes = [
+        hash_tree_root(randao_reveal_slashing.epoch),
+        randao_reveal_slashing.mask,
+    ]
+
+    assert bls_verify_multiple(
+        pubkeys=pubkeys,
+        message_hashes=message_hashes,
+        signature=randao_reveal_slashing.reveal,
+        domain=get_domain(
+            fork=state.fork,
+            epoch=randao_reveal_slashing.epoch,
+            domain_type=DOMAIN_RANDAO,
+        ),
+    )
+
+    assert randao_reveal_slashing.epoch > get_current_epoch(state)
```

Comment:
Do we want to allow for slashing in the case that it is just from 1 epoch in the future? Without a buffer, it is feasible that minor malicious reorgs could potentially be successful in including slashings. 

Say for example 64 builds on 62. Then 63 attempts to build on 62. 63 can then include the reveal from 64 as a slashing. Although they are likely to not make it into the canonical chain, times of medium latency might make the chances of success here non-negligible.

I would consider adding a constant `RANDAO_SLASHING_EPOCHS` or something similar that allows us to tune the constant via `assert randao_reveal_slashing.epoch > get_current_epoch(state) + RANDAO_SLASHING_EPOCHS` where `RANDAO_SLASHING_EPOCHS >= 1`